extensions/rich_text_components/Video/directives/oppia-noninteractive-video.component.ts

```diff
@@ -0,0 +1,121 @@
+// Copyright 2014 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Directive for the Video rich-text component.
+ *
+ * IMPORTANT NOTE: The naming convention for customization args that are passed
+ * into the directive is: the name of the parameter, followed by 'With',
+ * followed by the name of the arg.
+ */
+
+import { Component, ElementRef, Input, OnInit } from '@angular/core';
+import { downgradeComponent } from '@angular/upgrade/static';
+import { AutoplayedVideosService } from 'services/autoplayed-videos.service';
+import { ContextService } from 'services/context.service';
+import { HtmlEscaperService } from 'services/html-escaper.service';
+import { ServicesConstants } from 'services/services.constants';
+
+let apiLoaded = false;
+@Component({
+  selector: 'oppia-noninteractive-video',
+  templateUrl: './video.component.html',
+  styleUrls: []
+})
+export class NoninteractiveVideo implements OnInit {
+  @Input() autoplayWithValue: string;
+  @Input() endWithValue: string;
+  @Input() startWithValue: string;
+  @Input() videoIdWithValue: string;
+
+  playerVars = {
+    autoplay: 0,
+    origin: 'http://localhost:8181'
+  };
+  start: number;
+  end: number;
+  tabIndexVal: number;
+  videoId: string;
+  width: number;
+
+  constructor(
+    private autoplayedVideosService: AutoplayedVideosService,
+    private contextService: ContextService,
+    private el: ElementRef,
+    private htmlEscaperService: HtmlEscaperService
+  ) {}
+
+  ngOnInit(): void {
+    if (!apiLoaded) {
+      // This code loads the IFrame Player API code asynchronously, according to
+      // the instructions at
+      // https://developers.google.com/youtube/iframe_api_reference#Getting_Started
+      const tag = document.createElement('script');
+      tag.src = 'https://www.youtube.com/iframe_api';
+      document.body.appendChild(tag);
+      apiLoaded = true;
+    }
+    const start = this.htmlEscaperService.escapedJsonToObj(
+      this.startWithValue) as string;
+    const end = this.htmlEscaperService.escapedJsonToObj(
+      this.endWithValue) as string;
+    this.start = +start;
+    this.end = +end;
```

Comment:
```suggestion
    this.start = start;
    this.end = end;
```

Reply:
`start` and `end` are strings while `this.start` and `this.end` are numbers. We need this type casting.

Reply:
Should we do explicit type casting then? Like `Number(start)`?

Reply:
Or maybe if you think `Number` is too big refactor just add a comment.

Reply:
I think that Number() is the same as using `+`. I really don't see any difference.
![image](https://user-images.githubusercontent.com/23002114/119409157-7e373c00-bd04-11eb-96e8-170e5d10c9b6.png)

Reply:
Yeah, but what I mean is that (at least for me) `Number(x)` seems more readable and understandable than `+x`.

Reply:
Oh... for readability. Changed.